### DISCUSSION TITLE
[@types/quill]

### DISCUSSION THREAD

**Discussion by danielnixon** [DefinitelyTyped/DefinitelyTyped#56477] [Issues with a @types package]
In the @types/quill package (https://www.npmjs.com/package/@types/quill) version 1.3.10, the `Key` interface is missing a number of properties.

It is defined as:

```ts
export interface Key {
    key: string;
    shortKey?: boolean;
}
```

But if you have a look at the docs, you'll see, `altKey`, `ctrlKey`, `shiftKey`, etc are missing:

See: https://quilljs.com/docs/modules/keyboard/

[a comment by an automated account is not shown]

**Comment by danielnixon** [comment]
I see those props are in the 2.x branch... maybe I just have to bite the bullet and upgrade.

**Reply by danielnixon** [reply]
Upgrading is blocked for me. See: https://github.com/KillerCodeMonkey/ngx-quill/issues/1388

**Comment by 43081j** [comment]
they still have 1.x as their 'stable' published build which is why the types haven't yet been caught up to whats in 2.x. they've had 2.x in prerelease for so, so long though that im wondering if they'll ever get around to publishing it or if its just gone stale (or the opposite, neverending, blown out of proportion).

**Comment by danielnixon** [comment]
Those properties I refer to exist in the 1.x branch though, so they should be included in the 1.x types as well as the 2.x types? See e.g. https://github.com/quilljs/quill/blob/9cf0285caa6514356a7bba9db132ec7229eb254c/docs/1.0/docs/modules/keyboard.md

I guess the stalled 2.x release is even more reason to update the types in the 1.x branch?

**Reply by 43081j** [reply]
ah you are correct. and yeah possibly, 2.x has been in prerelease for so long maybe its widely used enough at this point that its effectively released. but the default npm tag is still 1.x by the looks of it (`npm i quill`)

**Reply by danielnixon** [reply]
Yeah, keeping the default at 1.x seems fine to me. It would be great to make sure the 1.x types have those keyboard properties though. I'll find time to create a PR in the coming days if nobody beats me :)
